swap_meet/vendor.py

```diff
@@ -1,2 +1,60 @@
+from .item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory = None ):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, inventory_item):
+        self.inventory.append(inventory_item)
+        return inventory_item
+
+    def remove(self, inventory_item):
+        if inventory_item in self.inventory:
+            self.inventory.remove(inventory_item)
+            return inventory_item
+        else:
+            return False
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items 
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            other_vendor.add(my_item)
+            self.remove(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            return True
+        else:
+            return False 
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        else:
+            self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
Nice reuse of `swap_items`.

Notice, though, that by using `swap_items`, this winds up being O(n) (mostly due to removing the items from the inventory). We could simply swap the items in place here to have O(1) runtime with something like:
```python
        self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]
```

Alternatively, we could think about how to modify the implementation of `swap_items` so that it would still work in general, but also give us O(1) in this specific case. Try that out if that sounds interesting!